src/sage/algebras/commutative_dga.py

```diff
@@ -670,11 +670,11 @@ def differential_matrix_multigraded(self, n, total=False):
         n = G(vector(n))
         dom = A.basis(n)
         cod = A.basis(n + self._degree_of_differential)
-        cokeys = [next(iter(a.lift().dict().keys())) for a in cod]
+        cokeys = [next(iter(a.lift().monomial_coefficients().keys())) for a in cod]
```

Comment:
Should I simplify this to
```suggestion
        cokeys = [next(iter(a.monomial_coefficients())) for a in cod]
```

Reply:
(similarly, below and throughout)

Reply:
I wouldn't; see the previous comment about `lift()`.